WIP: Updating Mapknitter to Rails 6.0.0

alaxalves opened the pull request:
Fixes #603 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkbox write 'x' within the square brackets)

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

alaxalves commented:
In order to get this running we'll need a ruby version > 2.5. What's your input on this? @sashadev-sky @jywarren

jywarren commented:
Hi @alaxalves - @icarito noted that Rails 6 is currently beyond Debian
stable's version, and i was wondering if we should start with a PR that
addresses deprecations and warnings without actually initiating the 6.0
upgrade just yet. Do we have an idea of Debian stable's schedule? I just
want to be careful about this and think well ahead, and I don't quite have
all the info to provide good input yet. Thanks so much!

On Tue, Jul 9, 2019 at 12:08 PM Álax de Carvalho Alves <
notifications@github.com> wrote:

> Fixes #603 <https://github.com/publiclab/mapknitter/issues/603> (<=== Add
> issue number here)
>
> Make sure these boxes are checked before your pull request (PR) is ready
> to be reviewed and merged. Thanks!
>
>    - PR is descriptively titled 📑 and links the original issue above 🔗
>    - tests pass -- look for a green checkbox ✔️ a few minutes after
>    opening your PR -- or run tests locally with rake test
>    - code is in uniquely-named feature branch and has no merge conflicts
>    📁
>    - screenshots/GIFs are attached 📎 in case of UI updation
>    - ask @publiclab/mapknitter-reviewers for help, in a comment below
>
> We're happy to help you get this ready -- don't be afraid to ask for help,
> and *don't be discouraged* if your tests fail at first!
>
> If tests do fail, click on the red X to learn why by reading the logs.
>
> Please be sure you've reviewed our contribution guidelines at
> https://publiclab.org/contributing-to-public-lab-software
>
> Thanks!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/publiclab/mapknitter/pull/824
> Commit Summary
>
>    - Bumping rails version
>
> File Changes
>
>    - *M* Gemfile
>    <https://github.com/publiclab/mapknitter/pull/824/files#diff-0> (4)
>    - *M* Gemfile.lock
>    <https://github.com/publiclab/mapknitter/pull/824/files#diff-1> (131)
>
> Patch Links:
>
>    - https://github.com/publiclab/mapknitter/pull/824.patch
>    - https://github.com/publiclab/mapknitter/pull/824.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/824?email_source=notifications&email_token=AAAF6J6XGCTGN7NXDOO4KJTP6SZWVA5CNFSM4H7HD3Y2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4G6ESRFQ>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J7RBR5CLPG4LNUPQFLP6SZWVANCNFSM4H7HD3YQ>
> .
>

jywarren commented:
Ah, interesting. Would upgrading to Ruby 2.5 be its own PR perhaps, that
could be done before Rails 6?

On Tue, Jul 9, 2019 at 12:12 PM Jeffrey Warren <jeff@unterbahn.com> wrote:

> Hi @alaxalves - @icarito noted that Rails 6 is currently beyond Debian
> stable's version, and i was wondering if we should start with a PR that
> addresses deprecations and warnings without actually initiating the 6.0
> upgrade just yet. Do we have an idea of Debian stable's schedule? I just
> want to be careful about this and think well ahead, and I don't quite have
> all the info to provide good input yet. Thanks so much!
>
> On Tue, Jul 9, 2019 at 12:08 PM Álax de Carvalho Alves <
> notifications@github.com> wrote:
>
>> Fixes #603 <https://github.com/publiclab/mapknitter/issues/603> (<===
>> Add issue number here)
>>
>> Make sure these boxes are checked before your pull request (PR) is ready
>> to be reviewed and merged. Thanks!
>>
>>    - PR is descriptively titled 📑 and links the original issue above 🔗
>>    - tests pass -- look for a green checkbox ✔️ a few minutes after
>>    opening your PR -- or run tests locally with rake test
>>    - code is in uniquely-named feature branch and has no merge conflicts
>>    📁
>>    - screenshots/GIFs are attached 📎 in case of UI updation
>>    - ask @publiclab/mapknitter-reviewers for help, in a comment below
>>
>> We're happy to help you get this ready -- don't be afraid to ask for
>> help, and *don't be discouraged* if your tests fail at first!
>>
>> If tests do fail, click on the red X to learn why by reading the logs.
>>
>> Please be sure you've reviewed our contribution guidelines at
>> https://publiclab.org/contributing-to-public-lab-software
>>
>> Thanks!
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>   https://github.com/publiclab/mapknitter/pull/824
>> Commit Summary
>>
>>    - Bumping rails version
>>
>> File Changes
>>
>>    - *M* Gemfile
>>    <https://github.com/publiclab/mapknitter/pull/824/files#diff-0> (4)
>>    - *M* Gemfile.lock
>>    <https://github.com/publiclab/mapknitter/pull/824/files#diff-1> (131)
>>
>> Patch Links:
>>
>>    - https://github.com/publiclab/mapknitter/pull/824.patch
>>    - https://github.com/publiclab/mapknitter/pull/824.diff
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/publiclab/mapknitter/pull/824?email_source=notifications&email_token=AAAF6J6XGCTGN7NXDOO4KJTP6SZWVA5CNFSM4H7HD3Y2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4G6ESRFQ>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAAF6J7RBR5CLPG4LNUPQFLP6SZWVANCNFSM4H7HD3YQ>
>> .
>>
>

alaxalves commented:
> Ah, interesting. Would upgrading to Ruby 2.5 be its own PR perhaps, that could be done before Rails 6?
> […](#)
> On Tue, Jul 9, 2019 at 12:12 PM Jeffrey Warren ***@***.***> wrote: Hi @alaxalves - @icarito noted that Rails 6 is currently beyond Debian stable's version, and i was wondering if we should start with a PR that addresses deprecations and warnings without actually initiating the 6.0 upgrade just yet. Do we have an idea of Debian stable's schedule? I just want to be careful about this and think well ahead, and I don't quite have all the info to provide good input yet. Thanks so much! On Tue, Jul 9, 2019 at 12:08 PM Álax de Carvalho Alves < ***@***.***> wrote: > Fixes #603 <#603> (<=== > Add issue number here) > > Make sure these boxes are checked before your pull request (PR) is ready > to be reviewed and merged. Thanks! > > - PR is descriptively titled  and links the original issue above  > - tests pass -- look for a green checkbox  a few minutes after > opening your PR -- or run tests locally with rake test > - code is in uniquely-named feature branch and has no merge conflicts >   > - screenshots/GIFs are attached  in case of UI updation > - ask @publiclab/mapknitter-reviewers for help, in a comment below > > We're happy to help you get this ready -- don't be afraid to ask for > help, and *don't be discouraged* if your tests fail at first! > > If tests do fail, click on the red X to learn why by reading the logs. > > Please be sure you've reviewed our contribution guidelines at > https://publiclab.org/contributing-to-public-lab-software > > Thanks! > ------------------------------ > You can view, comment on, or merge this pull request online at: > > #824 > Commit Summary > > - Bumping rails version > > File Changes > > - *M* Gemfile > <https://github.com/publiclab/mapknitter/pull/824/files#diff-0> (4) > - *M* Gemfile.lock > <https://github.com/publiclab/mapknitter/pull/824/files#diff-1> (131) > > Patch Links: > > - https://github.com/publiclab/mapknitter/pull/824.patch > - https://github.com/publiclab/mapknitter/pull/824.diff > > — > You are receiving this because you are subscribed to this thread. > Reply to this email directly, view it on GitHub > <#824?email_source=notifications&email_token=AAAF6J6XGCTGN7NXDOO4KJTP6SZWVA5CNFSM4H7HD3Y2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4G6ESRFQ>, > or mute the thread > <https://github.com/notifications/unsubscribe-auth/AAAF6J7RBR5CLPG4LNUPQFLP6SZWVANCNFSM4H7HD3YQ> > . >

Yes, I could open a MR upgrading to latest ruby. BTW can I push Rails 5 to unstable?

jywarren commented:
for sure push to unstable! Thanks!

On Tue, Jul 9, 2019 at 5:29 PM Álax de Carvalho Alves <
notifications@github.com> wrote:

> Ah, interesting. Would upgrading to Ruby 2.5 be its own PR perhaps, that
> could be done before Rails 6?
> … <#m_-5784711317799686700_>
> On Tue, Jul 9, 2019 at 12:12 PM Jeffrey Warren *@*.*> wrote: Hi
> @alaxalves <https://github.com/alaxalves> - @icarito
> <https://github.com/icarito> noted that Rails 6 is currently beyond Debian
> stable's version, and i was wondering if we should start with a PR that
> addresses deprecations and warnings without actually initiating the 6.0
> upgrade just yet. Do we have an idea of Debian stable's schedule? I just
> want to be careful about this and think well ahead, and I don't quite have
> all the info to provide good input yet. Thanks so much! On Tue, Jul 9, 2019
> at 12:08 PM Álax de Carvalho Alves < @.*> wrote: > Fixes #603
> <https://github.com/publiclab/mapknitter/issues/603> <#603
> <https://github.com/publiclab/mapknitter/issues/603>> (<=== > Add issue
> number here) > > Make sure these boxes are checked before your pull request
> (PR) is ready > to be reviewed and merged. Thanks! > > - PR is
> descriptively titled and links the original issue above > - tests pass --
> look for a green checkbox a few minutes after > opening your PR -- or run
> tests locally with rake test > - code is in uniquely-named feature branch
> and has no merge conflicts > > - screenshots/GIFs are attached in case of
> UI updation > - ask @publiclab/mapknitter-reviewers
> <https://github.com/orgs/publiclab/teams/mapknitter-reviewers> for help,
> in a comment below > > We're happy to help you get this ready -- don't be
> afraid to ask for > help, and *don't be discouraged* if your tests fail
> at first! > > If tests do fail, click on the red X to learn why by reading
> the logs. > > Please be sure you've reviewed our contribution guidelines at
> > https://publiclab.org/contributing-to-public-lab-software > > Thanks! >
> ------------------------------ > You can view, comment on, or merge this
> pull request online at: > > #824
> <https://github.com/publiclab/mapknitter/pull/824> > Commit Summary > > -
> Bumping rails version > > File Changes > > - *M* Gemfile >
> https://github.com/publiclab/mapknitter/pull/824/files#diff-0 (4) > - *M*
> Gemfile.lock >
> https://github.com/publiclab/mapknitter/pull/824/files#diff-1 (131) > >
> Patch Links: > > - https://github.com/publiclab/mapknitter/pull/824.patch
> > - https://github.com/publiclab/mapknitter/pull/824.diff > > — > You are
> receiving this because you are subscribed to this thread. > Reply to this
> email directly, view it on GitHub > <#824
> <https://github.com/publiclab/mapknitter/pull/824>?email_source=notifications&email_token=AAAF6J6XGCTGN7NXDOO4KJTP6SZWVA5CNFSM4H7HD3Y2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4G6ESRFQ>,
> > or mute the thread >
> https://github.com/notifications/unsubscribe-auth/AAAF6J7RBR5CLPG4LNUPQFLP6SZWVANCNFSM4H7HD3YQ
> > . >
>
> Yes, I could open a MR upgrading to latest ruby. BTW can I push Rails 5 to
> unstable?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/824?email_source=notifications&email_token=AAAF6J3DE2OH6RSNXDETANDP6T5PJA5CNFSM4H7HD3Y2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZRRXPA#issuecomment-509811644>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J3MHHPWUQEE6UXKHETP6T5PJANCNFSM4H7HD3YQ>
> .
>

kaustubh-nair commented on app/assets/javascripts/uploads.js:
```diff
@@ -5,7 +5,7 @@
 
 // iframe-transport is ajax file upload support for IE, Uncomment if needed      
-// javascript_include_tag  (file_upload + "js/jquery.iframe-transport")      
+// javascript_pack_tag  (file_upload + "js/jquery.iframe-transport")      
```
This is not working.
I get `ActionView::Template::Error: undefined method `javascript_pack_tag'`

alaxalves replied:
I know, the webpacker integration is not working.

kaustubh-nair replied:
There's a problem with the installation itself. The rake task doesn't work after adding `webpacker` to the gemfile.

alaxalves commented:
@kaustubh-nair If we want to upgrade to Rails 6 is highly recommended to use Webpacker. We should not revert as in 730774c. Actually we should try to integrate it properly by deprecating Sprockets usage. Please check out the following:
https://www.botreetechnologies.com/blog/introducing-jquery-in-rails-6-using-webpacker
https://medium.com/@coorasse/goodbye-sprockets-welcome-webpacker-3-0-ff877fb8fa79

kaustubh-nair commented:
@alaxalves I understand that. I just wanted to make sure everything is working properly before integrating webpacker. Webpacker was not working anyway so no harm in anything.
We can add it properly now. 
The problem is the installation files are not being loaded after installing the gem.
Running `bundle exec rake webpacker:install` doesn't work. Let's install it first before making changes to the code

alaxalves commented:
@kaustubh-nair I think it's time to rebase this with development, before https://github.com/publiclab/mapknitter/pull/805 gets merged.

kaustubh-nair commented:
@alaxalves I merged `development` here but there is some syntax error in `app/views/front_ui/index.html.erb` which  I can't figure out.

kaustubh-nair commented:
This occurs even after I copied `front_ui/index.html.erb` from `development`..weird

alaxalves commented:
Thanks, I'll check it out. How's system tests on Rails 5 doing @kaustubh-nair, do you need any help?

sashadev-sky commented:
@alaxalves from your comment "setting Webpacker with Rails 6.0.0 on #824 ?? I'm getting trouble trying to get the assets to precompile along with Webpack, since we dropped Sprockets."

What is the problem trying to fix and how can I reproduce it? If its referring to paper trail I think @jywarren has resolved this! I can link the PR if needed

alaxalves commented:
> @alaxalves from your comment "setting Webpacker with Rails 6.0.0 on #824 ?? I'm getting trouble trying to get the assets to precompile along with Webpack, since we dropped Sprockets."
> 
> What is the problem trying to fix and how can I reproduce it? If its referring to paper trail I think @jywarren has resolved this! I can link the PR if needed

Thx for your input @sashadev-sky. You reproduce it by running `rails assets:precompile`. If you use docker you have to rebuild your image, since we changed the ruby version. If u use RVM or similar you have to use ruby 2.6.3. If you're able to get the assets to precompile it would be great, because we would be using the newest of the newest and Mapknitter would be one the first projects in the world to fully deprecate Sprockets and fully use Webpacker(For JS, SCSS and images), a huge upgrade BTW. I have changed the entire directory structure for assets, as you'll notice. You'll get more info on the [Webpacker gem page](https://github.com/rails/webpacker). Maybe @jywarren could give us help on this thing too.
If this takes too much time or anything like this, I think the best option is dropping this Webpacker integration for now, and set Sprockets back again(since it was working fine and all). Then get Rails 6 stable and open up an issue regarding this Webpacker thing or anything like it.
I have been trying to get those assets precompiled for the entire past week, but no success since I not very into the Rails' asset pipeline. Maybe we could set an ending date to try this, since the end of GSoC is coming to an end and I'd like the deliver something stable for u guys. What do you think? @jywarren @sashadev-sky

jywarren commented:
Gosh, i mean webpacker instead of sprockets sounds really nice. It's hard to believe Sprockets is still being widely used in Rails, but many of the asset pipeline components seem brittle when we try to upgrade them (like the Yarn integration), in my experience. I think it's totally OK to split them, and defer to you on this. 

Indeed although webpacker would be really nice, and i totally appreciate your work on it, there are other issues I feel are more urgent - like fixing our coverage tracking (codecov or otherwise) and getting system tests in place to help us merge more JS dependabot updates. So, I still defer, but those are the things in mind for me.

jywarren commented:
Hi, @alaxalves -- this is so exciting! Did you have a chance to look into the Debian Rails 6 schedule, or do some thinking/planning on the potential issues related to getting ahead of Debian? Let's also ping @icarito on this topic!

alaxalves commented:
@icarito @jywarren According to this link they have only mapped the upcoming functionalities for Rails 6. But I have found no info on release dates. @arthurbdiniz Do you have any info on this?

alaxalves commented on Gemfile:
```diff
@@ -39,7 +43,7 @@ group :dependencies do
 
   # Process manager for applications with multiple components
-  gem 'foreman', '~> 0.85.0'
+  gem 'foreman', git: 'https://github.com/alaxalves/foreman.git', branch: 'master'
```
Had to do this to keep the `thor` gem compatibility both with `rails` and `foreman` gems. Done [here](https://github.com/alaxalves/foreman/commit/ac1c0919f8a8f5c81112bc1bb9b41a2565d56202)

arthurbdiniz commented:
> @icarito @jywarren According to [this link](https://wiki.debian.org/Teams/Ruby/Rails6/RailsBreakingChanges) they have only mapped the upcoming functionalities for Rails 6. But I have found no info on release dates. @arthurbdiniz Do you have any info on this?

Hi everyone, I did some research here and I have some information for you guys.

The current rails version on Debian is `5.2.2`. And all this information can be found on https://tracker.debian.org/pkg/rails

So I saw that rails `6.0.0` still not stable yet looking at https://github.com/rails/rails/releases
The latest stable is `5.2.3` and the Debian package maintainer is already in the process of upgrading, see https://salsa.debian.org/ruby-team/rails. 

To the maintainer start the packaging of the rails `6.0.0` first this version needs to be on stable.

Since rails are within the `Debian Ruby ​​Team` and when rails come to a new stable release the maintainer is usually advised to make a new version of the package. 


In case the maintainer disappears and show no sign of the new version, there are two ways to contact the maintainer by sending an email to the team list pkg-ruby-extras-maintainers@lists.alioth.debian.org or sending an email directly to the maintainer `Utkarsh Gupta` guptautkarsh2102@gmail.com, which is my friend 😄, showing interest in packaging the new version. 

⚠️ Remembering that we at Debian do as volunteer work and I recommend waiting a while before sending messages to the maintainer.

#### Extra
You can also find him on `Debian IRC`, the hostname is `irc.debian.org` which is an alias for `irc.oftc.net` on channel `#debian-ruby` or directly to him with the nickname `utkarsh2102[m]`

Hope this helps!

alaxalves commented:
> > @icarito @jywarren According to [this link](https://wiki.debian.org/Teams/Ruby/Rails6/RailsBreakingChanges) they have only mapped the upcoming functionalities for Rails 6. But I have found no info on release dates. @arthurbdiniz Do you have any info on this?
> 
> Hi everyone, I did some research here and I have some information for you guys.
> 
> The current rails version on Debian is `5.2.2`. And all this information can be found on https://tracker.debian.org/pkg/rails
> 
> So I saw that rails `6.0.0` still not stable yet looking at https://github.com/rails/rails/releases
> The latest stable is `5.2.3` and the Debian package maintainer is already in the process of upgrading, see https://salsa.debian.org/ruby-team/rails.
> 
> To the maintainer start the packaging of the rails `6.0.0` first this version needs to be on stable.
> 
> Since rails are within the `Debian Ruby ​​Team` and when rails come to a new stable release the maintainer is usually advised to make a new version of the package.
> 
> In case the maintainer disappears and show no sign of the new version, there are two ways to contact, by sending an email to the team list [pkg-ruby-extras-maintainers@lists.alioth.debian.org](mailto:pkg-ruby-extras-maintainers@lists.alioth.debian.org) or sending an email directly to the maintainer `Utkarsh Gupta` [guptautkarsh2102@gmail.com](mailto:guptautkarsh2102@gmail.com), which is my friend , showing interest in packaging the new version.
> 
>  Remembering that we at Debian do as volunteer work and I recommend waiting a while before sending messages to the maintainer.
> 
> #### Extra
> You can also find him on `Debian IRC`, the hostname is `irc.debian.org` which is an alias for `irc.oftc.net` on channel `#debian-ruby` or directly to him with the nickname `utkarsh2102[m]`
> 
> Hope this helps!

Thx a lot my dude! This was super helpful! Check this out @jywarren @icarito

jywarren commented:
Thank you so much, super helpful! Echoing something from @alaxalves from our chatroom:

>  Rails 6 has just been shipped
> I think we'll soon have it packed on Debian
> I'll update the merge
> https://edgeguides.rubyonrails.org/6_0_release_notes.html
> https://weblog.rubyonrails.org/2019/8/15/Rails-6-0-final-release/

😄

kaustubh-nair commented:
Hi @alaxalves has there been any updates on the Debian release?

alaxalves commented:
> Hi @alaxalves has there been any updates on the Debian release?

Not yet, last pkg release has been for Rails 5.2.3 on October 2019.

alaxalves commented:
@jywarren Good news, seems like we're close to a stable Rails 6.0 package. I'll make sure to update this ASAP
